### DISCUSSION TITLE
Allow multiple treesitter languages?

### DISCUSSION THREAD

**Discussion by flippette** [helix-editor/helix#9219] [Ideas]
Currently, the `view!` macro in `leptos` makes helix highlighting very weird:

![image](https://github.com/helix-editor/helix/assets/43138244/c060cee8-73f2-471f-aee4-312407daeeee)

Is it possible to make treesitter use a different grammar inside the `view!` macro? Or in a more general sense?

**Comment by kirawi** [comment]
You could modify https://github.com/helix-editor/helix/blob/master/runtime/queries/rust/injections.scm locally to support HTML injections for the `view` macro, but the caveat is that it Leptos mixes it with Rust syntax so it wouldn't be correct anyway.

**Comment by erasin** [comment]
https://github.com/helix-editor/helix/discussions/9993 with tree-sitter-rshtml
